### PR TITLE
Copies phases temporal memory implementation to nupic.research 

### PR DESCRIPTION
**Pull request by andrewmalta13**
Nupic's temporal memory will be changing soon when <a href="https://github.com/numenta/nupic/pull/3197"> this </a> pull request is merged. This will cause extended temporal memory and faulty temporal memory to fail as it changes the exposed api. This pr is to copy the old python implementation to nupic research for the two files to continue to extend.

Extended temporal memory is being changed on the c++ side and a python port will probably follow, making this a somewhat temporary change for etm.

Faulty temporal memory may just continue to extend this copy of the phases temporal memory.

As discussed:
@scottpurdy @mrcslws

### CODE REVIEW

File: htmresearch/algorithms/temporal_memory_phases.py [hunk 1]
```diff
@@ -0,0 +1,833 @@
...
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+A deprectated version of Temporal Memory implementation in Python. Uses
```
Comment (mrcslws) [inline htmresearch/algorithms/temporal_memory_phases.py]: Typo: deprecated

File: htmresearch/algorithms/temporal_memory_phases.py [hunk 2]
```diff
@@ -0,0 +1,833 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2014, Numenta, Inc.  Unless you have an agreement
```
Comment (mrcslws) [inline htmresearch/algorithms/temporal_memory_phases.py]: Might as well change this to 2014-2016

**Comment by mrcslws** [comment]
Looks good. It's ready for you, @scottpurdy 

It'd be good for us to merge this before merging the NuPIC TM change so these two nupic.research TM subclasses are never broken.

**Comment by andrewmalta13** [comment]
@scottpurdy nupic.research extendedTemporalMemory and faultyTemporalMemory are broken until this is merged.

File: htmresearch/algorithms/faulty_temporal_memory.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 """
 import numpy
 from collections import defaultdict
-from nupic.research.temporal_memory import TemporalMemory
+from temporal_memory_phases import TemporalMemory
```
Comment (scottpurdy) [inline htmresearch/algorithms/faulty_temporal_memory.py]: Can you do a fully qualified import here? Like `from htmresearch.algorithms.temporal_memory_phases import TemporalMemory`?

The implicit relative import style that you use here is problematic. There is an explicit relative import using a `.` prefix but absolute imports are the best option in most cases.

**Comment by scottpurdy** [comment]
👍

**Pull request merged**